cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/Actions.hs

```diff
@@ -376,3 +379,41 @@ runGovernanceActionTreasuryWithdrawalCmd
   firstExceptT GovernanceActionsCmdWriteFileError . newExceptT
     $ conwayEraOnwardsConstraints eon
     $ writeFileTextEnvelope outFile Nothing proposal
+
+runGovernanceActionHardforkInitCmd :: ()
+  => GovernanceActionHardforkInitCmdArgs era
+  -> ExceptT GovernanceActionsError IO ()
+runGovernanceActionHardforkInitCmd
+    Cmd.GovernanceActionHardforkInitCmdArgs
+      { Cmd.eon
+      , Cmd.networkId
+      , Cmd.deposit
+      , Cmd.returnStakeAddress
+      , Cmd.mPrevGovernanceActionId
+      , Cmd.proposalUrl
+      , Cmd.proposalHash
+      , Cmd.protVer
+      , Cmd.outFile
+      } = do
+  depositStakeCredential
+    <- firstExceptT GovernanceActionsReadStakeCredErrror
+         $ getStakeCredentialFromIdentifier returnStakeAddress
+
+  let proposalAnchor = L.Anchor
+        { L.anchorUrl = unProposalUrl proposalUrl
+        , L.anchorDataHash = proposalHash
+        }
+
+  let sbe = conwayEraOnwardsToShelleyBasedEra eon
+      govActIdentifier = L.maybeToStrictMaybe
+                           $ shelleyBasedEraConstraints sbe
+                           $ uncurry createPreviousGovernanceActionId <$> mPrevGovernanceActionId
+      initHardfork = InitiateHardfork
+                      govActIdentifier
+                      protVer
+
+      proposalProcedure = createProposalProcedure sbe networkId deposit depositStakeCredential initHardfork proposalAnchor
+
+  firstExceptT GovernanceActionsCmdWriteFileError . newExceptT
+    $ conwayEraOnwardsConstraints eon
+    $ writeFileTextEnvelope outFile Nothing proposalProcedure
```

Comment:
Could we replace the `Nothing` by something, so that the generated proposal has an helpful description text?